src/client/dfuse/ops/lookup.c

```diff
@@ -250,15 +241,24 @@ dfuse_cb_lookup(fuse_req_t req, struct dfuse_inode_entry *parent,
 		DFUSE_TRA_DEBUG(ie,
 				"check_for_uns_ep() returned %d", rc);
 		if (rc != 0)
-			D_GOTO(err, rc);
+			D_GOTO(out_release, rc);
 	}
 
 	dfuse_reply_entry(fs_handle, ie, NULL, false, req);
 	return;
 
-err:
-	DFUSE_REPLY_ERR_RAW(fs_handle, req, rc);
+out_release:
 	dfs_release(ie->ie_obj);
-free:
+out_free:
 	D_FREE(ie);
+out:
+	if (rc == ENOENT && parent->ie_dfs->dfs_attr_timeout > 0) {
```

Comment:
This particular part is re-written as part of https://github.com/daos-stack/daos/pull/4540 anyway.